flink-connector-gcp-pubsub/src/test/java/org/apache/flink/connector/gcp/pubsub/sink/table/CredentialProviderFactoryTest.java

```diff
@@ -0,0 +1,202 @@
+    static {
+        try {
+            // Generate a private key for testing
+            KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+            keyPairGenerator.initialize(2048); // Key size
+            KeyPair keyPair = keyPairGenerator.generateKeyPair();
+            PrivateKey privateKey = keyPair.getPrivate();
+            PKCS8EncodedKeySpec pkcs8EncodedKeySpec =
+                    new PKCS8EncodedKeySpec(privateKey.getEncoded());
+            PrivateKey key = KeyFactory.getInstance("RSA").generatePrivate(pkcs8EncodedKeySpec);
+            // private key
+            String testPkcs8PrivateKey =
+                    "-----BEGIN PRIVATE KEY-----\n"
+                            + Base64.getEncoder().encodeToString(key.getEncoded()) // private key
+                            + "\n-----END PRIVATE KEY-----";
+            TEST_JSON_CREDENTIALS =
+                    "{\"type\": \"service_account\", "
+                            + "\"project_id\": \"project-id\","
+                            + " \"private_key_id\": \"1234567890abcdef1234567890abcdef12345678\", "
+                            + "\"private_key\": \""
+                            + testPkcs8PrivateKey
+                            + "\", \"client_email\": \"\", \"client_id\": \""
+                            + TEST_CLIENT_ID
+                            + "\", "
+                            + "\"auth_uri\": \"https://accounts.google.com/o/oauth2/auth\","
+                            + " \"token_uri\": \"https://oauth2.googleapis.com/token\", "
+                            + "\"auth_provider_x509_cert_url\": \"auth-provider-x509-cert-url\", "
+                            + "\"client_x509_cert_url\": \"client-x509-cert-url\"}";
+        } catch (InvalidKeySpecException | NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void createDefaultCredentialsProvider() {
+        CredentialProviderFactory factory = new CredentialProviderFactory();
+        CredentialsProvider credentialsProvider =
+                factory.createCredentialsProvider(CredentialProviderType.DEFAULT, null);
+        assertThat(credentialsProvider).isNotNull();
+        assertThat(credentialsProvider).isInstanceOf(GoogleCredentialsProvider.class);
+    }
+
+    @Test
+    public void createGoogleCredentialsProvider() {
+        CredentialProviderFactory factory = new CredentialProviderFactory();
+        CredentialsProvider credentialsProvider =
+                factory.createCredentialsProvider(CredentialProviderType.GOOGLE_CREDENTIALS, null);
+        assertThat(credentialsProvider).isNotNull();
+        assertThat(credentialsProvider).isInstanceOf(GoogleCredentialsProvider.class);
+    }
+
+    @Test
+    public void createFixedCredentialsProviderFailsIfOptionsAreNull() {
+        CredentialProviderFactory factory = new CredentialProviderFactory();
+        assertThatExceptionOfType(NullPointerException.class)
```

Comment:
I'm not sure NPE is a good exception if user is able to see it...
is it possible to use something like `IlleagalArgumentException` with corresponding error message?